### PR TITLE
Add autocomplete API endpoint and UI

### LINKED ISSUE #292: Autocomplete on people, categories, object
We should implement autocomplete for...

- [ ] people
- [ ] categories
- [ ] object

..as a first pass on the search.

### PR DESCRIPTION
**Pull request by alanshaw**
* Adds API endpoint `GET /autocomplete?q=` (JSONAPI only)
* Autocompletes on field `summary_title`
* Debounces query by 500ms
* Configured to suggest on words with 3 characters or more

resolves #292

screen shot 2016-08-18 at 08 07 41

### CODE REVIEW

**Comment by alanshaw** [comment]
@SimonLab @Danwhy - would be good to get this in today ;P

**Comment by Danwhy** [comment]
@simonlab LGTM, feel free to merge if you're happy with it

File: lib/autocomplete.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  const searchOpts = {
+    searchName: 'autocomplete',
+    index: 'smg',
+    size: queryParams.size || 3,
+    _source: ['summary_title'],
+    body: {
+      query: {
+        bool: {
+          must,
```
Comment (SimonLab) [inline lib/autocomplete.js]: I think you don't need to precise the property must if it's empty, but is' more a stylish issue

**Comment by SimonLab** [comment]
:+1: LGTM

**Pull request merged**